### PR TITLE
Docs: Improvements for Installation, Index and Compiling

### PR DESCRIPTION
**Pull request by chgue**
### - What I did

1. Improved "Installing Viper"
1.1. Added Docker and online compiler
1.2. Restructured it a little bit
1.3. Added the commands needed to a compile a contract after the tests so users can verify that everything works correctly.
2. Added "Compiling a Contract"
Mostly copied from the readme. Changed wording and adapted the markup.
3. Improved Index
3.1 Added "Compiling a Contract" to the navigation menu
3.2 Improved structure and content
Changed the structure and incorporated parts of the readme. Reworded some things so that they fit better in the layout of the page.

Feedback greatly appreciated!

P.S. English isn't my native language so there might be (I hope not!) multiple errors and strange sounding sentences.

### - How I did it
With the powers of Ctrl-C, Ctrl-V, and a little bit of writing :wink:
### - How to verify it
Read it.
### - Description for the changelog
None needed I think?

### CODE REVIEW

**Comment by fubuloubu** [comment]
Quickly browsed through and the updated language is definitely clearer, which I like.

File: docs/installing-viper.rst [hunk 1]
```diff
@@ -99,3 +115,12 @@ install and test commands
     ::
         python setup.py install
         python setup.py test
+******
+Docker
+******
+A Docker image which executes most of the installation steps is provided.
+To ensure that everything works correctly after the installtion, please run the test commands
+and try compiling a contract:
+::
```
Comment (DavidKnott) [inline docs/installing-viper.rst]: It'd be great if you could explain how to setup docker starting with cloning the repo (e.i. `git clone ...`).

Comment (DavidKnott) [inline docs/installing-viper.rst]: It'd be great if you could explain how to setup docker starting with cloning the repo (e.i. `git clone ...`).

File: docs/index.rst [hunk 2]
```diff
@@ -3,32 +3,65 @@
...
+#####
 
-* Bounds and overflow checking, both on array accesses and on arithmetic
-* Support for signed integers and decimal fixed point numbers
-* Decidability - it's possible to compute a precise upper bound on the gas consumption of any function call
-* Strong typing, including support for units (eg. timestamp, timedelta, seconds, wei, wei per second, meters per second squared)
-* Maximally small and understandable compiler code size
-* Limited support for pure functions - anything marked constant is NOT allowed to change the state
+Viper is an **experimental**, contract-oriented, and high-level programming language whose syntax is similar to Python and it is
+designed to target the Etherum Virtual Machine (EVM).
```
Comment (DavidKnott) [inline docs/index.rst]: How do you think this sounds?
Viper is an **experimental**, contract-oriented, pythonic programming language that is designed to run on the Etherum Virtual Machine (EVM).

File: docs/compiling-a-contract.rst [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+.. note:: 
+    Since .vy is not official a language supported by any syntax highlighters or linters,
+    it is recommended to name your Viper file ending with `.v.py` in order to have Python syntax highlighting.
+
+An `online compiler <https://viper.tools/>`_ is available as well, which lets you experiment with
+the language without having to install Viper. The online compiler allows you to compile to ``bytecode`` and/or ``LLL``.
+
+.. note::
+    While the viper version of the online compiler is updated on a regular basis it might
+    be a bit behind the latest version found in the master branch of the repository.
```
Comment (DavidKnott) [inline docs/compiling-a-contract.rst]: It's great that you added this note :)

File: docs/installing-viper.rst (hunk 1 above)
Comment (chgue) [reply docs/installing-viper.rst]: @DavidKnott Sure, but I am not really familiar with Docker. After a quick search: `docker build https://github.com/ethereum/viper.git -t viper:v1` (Not sure if the :v1 is unnecessary but probably easier to keep track of future images v2, ...) and then  `docker run viper:v1`?

File: docs/compiling-a-contract.rst (hunk 3 above)
Comment (chgue) [reply docs/compiling-a-contract.rst]: @DavidKnott Yes definitely better. It's just adapted from the Solidity docs which isn't ideal in this case. I think I'll add a link to ethdocs.org as well.

File: docs/index.rst (hunk 2 above)
Comment (chgue) [reply docs/index.rst]: @DavidKnott Yes definitely better. It's just adapted from the Solidity docs which isn't ideal in this case. I think I'll add a link to ethdocs.org as well.

Comment (fubuloubu) [reply docs/index.rst]: :+1:

File: docs/installing-viper.rst [hunk 4]
```diff
@@ -99,3 +115,12 @@ install and test commands
     ::
         python setup.py install
         python setup.py test
+******
```
Comment (DavidKnott) [inline docs/installing-viper.rst]: What do you think of using make instead?  Changing `python setup.py install` to `make` and `python setup.py test` to `make test`.

File: docs/installing-viper.rst (hunk 1 above)
Comment (DavidKnott) [reply docs/installing-viper.rst]: @chgue These are the commands you need to run:
(put the `git clone viper` here)
1. `docker build -t viper .`
2. `docker run -it viper /bin/bash`
3. `viper examples/crowdfund.v.py`

Comment (chgue) [reply docs/installing-viper.rst]: @DavidKnott I tested the ` build https://github.com/ethereum/viper.git -t viper:v1` command  yesterday and it worked so I commited that. Do you want me to change it to your steps? 

Included the other changes you mentioned as well

**Comment by DavidKnott** [comment]
@chgue Looks good, rebase and I'll merge it in.

**Comment by chgue** [comment]
@DavidKnott Thank you! Rebased.

**Comment by DavidKnott** [comment]
Thanks a lot, merging it in :)

**Pull request merged**